exercises/accumulate/example.rs

```diff
@@ -1,18 +1,9 @@
-pub fn map_function(values: Vec<i32>, f: &Fn(i32) -> i32) -> Vec<i32> {
-    let mut out: Vec<i32> = vec![];
-    for val in values {
-        out.push(f(val))
-    }
-    out
-}
-
-pub fn map_closure<F>(values: Vec<i32>, f: F) -> Vec<i32>
+pub fn map<F>(mut values: Vec<i32>, f: F) -> Vec<i32>
```

Comment:
Interesting that it mutates `values`. On the one hand, this behaviour is unlike traditional `map`. On the other hand, the tests can't tell the difference, and they also do not preclude a non-mutating implementation. This is worth its own issue and not worth holding up this PR.

Reply:
> This is worth its own issue

https://github.com/exercism/rust/issues/628